### PR TITLE
562: Adding file payment consent tests

### PR DESCRIPTION
**Pull request by frlaura-jianu**
Issue: https://github.com/securebankingaccesstoolkit/securebankingaccesstoolkit/issues/562
Description: Added file payment consents functional tests including the ones for the new SubmitFile API (supporting both XML & JSON content-types).

### CODE REVIEW

File: src/test/kotlin/com/forgerock/uk/openbanking/support/payment/PaymentApiClient.kt [hunk 1]
```diff
@@ -85,6 +101,72 @@ class PaymentApiClient(val tpp: Tpp) {
             }
             return result.get()
         }
+
+        inline fun sendFileRequest(): Boolean {
+            if (jwsSignatureProducer != null && jsonBody != null) {
```
Comment (jorgesanchezperez) [inline src/test/kotlin/com/forgerock/uk/openbanking/support/payment/PaymentApiClient.kt:106]: It seems a duplicated method to me, the only difference is the content type, could be better create only one method and pass the content type as parameter like:
```kotlin
import org.apache.http.entity.ContentType
import com.github.kittinunf.fuel.core.Headers
sendFileRequest(contentType: ContentType) {  // ContentType.APPLICATION_JSON or ContentType.TEXT_XML
.....
request.header(Headers.CONTENT_TYPE, contentType)
}
````

**Review by jorgesanchezperez — CHANGES_REQUESTED** [review]
Avoid duplications.

File: src/test/kotlin/com/forgerock/uk/openbanking/support/payment/PaymentApiClient.kt (hunk 1 above)
Comment (frlaura-jianu) [reply src/test/kotlin/com/forgerock/uk/openbanking/support/payment/PaymentApiClient.kt:100]: We have refactored according to the requested changes.

**Review by jorgesanchezperez — APPROVED** [review]
Good Job!!

**Comment by frlaura-jianu** [comment]
> Good Job!!

Thanks !!

**Pull request merged**